libvast/include/vast/detail/generator.hpp

```diff
@@ -208,14 +209,12 @@ class [[nodiscard]] generator {
   generator(const generator& other) = delete;
 
   ~generator() {
-    if (m_coroutine)
+    if (m_coroutine) {
       m_coroutine.destroy();
+    }
   }
 
-  generator& operator=(generator other) noexcept {
-    swap(other);
-    return *this;
-  }
```

Comment:
It can result in an overload resolution failure:

```
$ cat assignment-operator.cpp 
#include <utility>

class A {
public:
	A() {}
	A(A&&) {}

	A& operator=(A other) { return *this; }
	A& operator=(A&& other) { return *this; }
};


int main() {
  A a1, a2;
  a1 = std::move(a2);
}
benno@bourbaki:~/code/cpp$ g++ assignment-operator.cpp 
assignment-operator.cpp: In function ‘int main()’:
assignment-operator.cpp:15:20: error: ambiguous overload for ‘operator=’ (operand types are ‘A’ and ‘std::remove_reference<A&>::type’ {aka ‘A’})
   15 |   a1 = std::move(a2);
      |                    ^
assignment-operator.cpp:8:12: note: candidate: ‘A& A::operator=(A)’
    8 |         A& operator=(A other) { return *this; }
      |            ^~~~~~~~
assignment-operator.cpp:9:12: note: candidate: ‘A& A::operator=(A&&)’
    9 |         A& operator=(A&& other) { return *this; }
      |            ^~~~~~~~
```

Reply:
Thanks, that makes sense.